src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { FC, useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: FC<Props> = ({ selectedTodo, setSelectedTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const { id, title, completed, userId } = selectedTodo;
```

Comment:
You are initializing your state with props. This is considered an anti-pattern in React. The reason is that it can lead to bugs that are very difficult to find. When React re-renders your component, it will keep the initial state of your component. This means that even if your props change, your state will not be updated. You can read more about it [here](https://stackoverflow.com/a/50403930), [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state) and [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state).